pkg/doc/signature/ed25519signature2018/suite.go

```diff
@@ -36,6 +42,89 @@ type signer interface {
 	Sign(data []byte) ([]byte, error)
 }
 
+// CryptoSigner defines a signer which uses crypto instance to sing the message using particular key handle.
+type CryptoSigner struct {
+	cr crypto.Crypto
+	kh interface{}
+}
+
+// NewCryptoSigner creates a new CryptoSigner.
+func NewCryptoSigner(cr crypto.Crypto, kh interface{}) *CryptoSigner {
+	return &CryptoSigner{
+		cr: cr,
+		kh: kh,
+	}
+}
+
+// Sign will sign document and return signature
+func (s CryptoSigner) Sign(msg []byte) ([]byte, error) {
+	return s.cr.Sign(msg, s.kh)
+}
+
+type verifier interface {
+	// Verify will verify a signature.
+	Verify(pubKeyValue interface{}, doc, signature []byte) error
+}
+
+// LocalCryptoVerifier defines a verifier which is based on Local KMS and Crypto
+// which uses keyset.Handle as input for verification.
+type LocalCryptoVerifier struct {
```

Comment:
This doesn't belong in this package @Baha-sk @kdimak

Reply:
Agree, this struct can do any signature verification, not just ed25519.

ed25519signaturesuite2018 can use this struct to do verification, as well as any NIST approved ecdsa signature suite.

Reply:
More specifically, this package should not know what kms is in use nor anything about tink.